doc/developer/design/20210831_correctness.md

```diff
@@ -0,0 +1,124 @@
+# <Insert Name of Feature>
+
+## Summary
+
+A description of the correctness properties Materialize provides and the terms and concepts we use to describe them.
+
+## Goals
+
+The main goal is to lay the foundation for ongoing work to clearly articulate correctness properties of Materialize.
+This includes the introduction of concepts that underlie correctess, the components that will enforce them, and how they can communicate clearly.
+
+## Non-Goals
+
+It is not a goal to prescribe fine detail in how each of the components should behave, or which should have certain responsibilities.
+It is not a goal to prescribe which behavior Materialize should support, but only how to speak about a "correct" subset of it.
+It is not a goal to advocate (much) for DEFINITE over NON-VOLATILE.
+
+## Description
+
+Materialize means to provide several correctness guarantees, which manifest as various consistency and durability properties.
+At the heart of this is the idea that the ordering of events can be explicitly resolved and recorded, after which queries and views over the data become a matter of correct evaluation on completed subsets of time.
+The guarantees of Materialize derive from the durability of these recorded data, and the determinism of views over them.
+
+The primary term introduced here is "definite" to modify a collection of data.
+A collection is definite if it is uniquely named or defined such that anyone with the same identification will see the same collection of data at the same logical times.
```

Comment:
what is "identification"

Reply:
I couldn't think of a good way to say "name or definition" in a way that wasn't .. well it felt almost tautological. But the intent is that one can identify collections either nominally ("source 419") or structurally ("this specific view over these sources"), and these identifications are sufficient to uniquely bind the collection of updates. Typing that out made it a bit clearer for me, so I'll add that unless further confusion has spilled out.

Reply:
This explanation makes sense. It's just name resolution.

Reply:
This seems to preclude ever doing compaction. If one party has a particular identification at a particular wall-clock time, and it returns a set of data for some logical time, then another party at a later wall-clock time should be able to recover that same set of data, according to my interpretation of the definition of "definite" as written here.

Reply:
Oh, I suppose I just needed to read a few more lines to see:

> A handle to a collection can be "definite from `since` until `upper`" (names pending) if it reflects the definite results for times greater or equal to `since` but not greater or equal to `upper`.

This addresses my concern.